src/plugins/data/common/datasets/types.ts

```diff
@@ -256,6 +258,29 @@ export interface DatasetField {
   // TODO:  osdFieldType?
 }
 
+export enum DatasetCanvasBannerComponentType {
+  CALLOUT = 'callout',
+  CUSTOM = 'custom',
+}
+
+export type DatasetCanvasBannerProps =
```

Comment:
i think we should have a `BaseDatasetCanvasBannerProps`  or `DatasetCanvasBannerProps` that has required props to add a new banner. Then create a type called `DatasetCanvasCalloutBannerProps` and `DatasetCanvasCustomBannerProps` or something.

if developing these union types become quiet difficult to understand sometimes and caused the linter to complain because the type isn't strict enough. easier to work with explicit types.

as far as the base props i think we should have at least 
```typescript
export type DatasetCanvasBannerProps = {
 // required. unique id, this will be used to associate a dataset to a canvas props. this is different than title because title is something that could be displayed but if i have two devs trying to add a new banner called `Index Callout Banner` (haven't looked at the code yet) but if the component doesn't change in an ID then maybe we won't render an update by accident.
  id: string;
  title: string; // required. going from optional to required would be a breaking change. and haven't looked at how this is used but i can imagine we don't really want to have a case where title is undefined.

Reply:
that or `title` to be `name` similar to the flyout stuff. 

we should also consider maybe `key` but we should def include
```
className?: string;
'data-test-subj'?: string;
```

writing tests on this abstract types would make this so much easier and if we have to style them the className will help